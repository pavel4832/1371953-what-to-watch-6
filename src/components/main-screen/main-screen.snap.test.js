import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import MainScreen from './main-screen';
import configureStore from 'redux-mock-store';
import MOCK_STORE from '../../mock/mock-store';

const mockStore = configureStore({});

it(`Should MainScreen render correctly`, () => {
  const store = mockStore(MOCK_STORE);
  const history = createMemoryHistory();
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <MainScreen />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
