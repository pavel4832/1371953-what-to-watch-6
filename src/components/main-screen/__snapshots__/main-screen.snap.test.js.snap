// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainScreen render correctly 1`] = `
<div>
  <section
    class="movie-card"
  >
    <div
      class="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      class="visually-hidden"
    >
      WTW
    </h1>
    <header
      class="page-header movie-card__head "
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      
      <h1
        class="page-title user-page__title"
        style="display: none;"
      />
      <div
        class="user-block"
      >
        <div
          class="user-block__avatar"
        >
          <a
            href="/mylist"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/1.png"
              width="63"
            />
          </a>
        </div>
        <a
          class="user-block__link"
          href="/"
        >
          Log out
        </a>
      </div>
    </header>
    <div
      class="movie-card__wrap"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Dramas
            </span>
            <span
              class="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              data-testid="play-button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  data-testid="add"
                  xlink:href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog"
    >
      <h2
        class="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        class="catalog__genres-list"
      >
        <li
          class="catalog__genres-item catalog__genres-item--active"
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            All genre
          </a>
        </li>
        <li
          class="catalog__genres-item "
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Comedies
          </a>
        </li>
        <li
          class="catalog__genres-item "
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Dramas
          </a>
        </li>
        <li
          class="catalog__genres-item "
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Sci-Fi
          </a>
        </li>
        <li
          class="catalog__genres-item "
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Thrillers
          </a>
        </li>
      </ul>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            This is mock VideoPlayer
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/2"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            This is mock VideoPlayer
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/3"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            This is mock VideoPlayer
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/4"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            This is mock VideoPlayer
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/5"
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            This is mock VideoPlayer
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/6"
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            This is mock VideoPlayer
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/7"
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            This is mock VideoPlayer
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/8"
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            This is mock VideoPlayer
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/9"
            >
              Johnny English
            </a>
          </h3>
        </article>
      </div>
      <div
        class="catalog__more"
      >
        <button
          class="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
